core/dbt/include/global_project/macros/materializations/materialized_view.sql

```diff
@@ -0,0 +1,68 @@
+{%- materialization materialized_view, default -%}
+
+    {%- set materialization = adapter.make_materialization_from_runtime_config(config, 'materialized_view') -%}
```

Comment:
This config variable is the one documented [here](https://docs.getdbt.com/reference/dbt-jinja-functions/config), correct? 
> The config variable exists to handle end-user configuration for custom [materializations](https://docs.getdbt.com/terms/materialization). Configs like unique_key can be implemented using the config variable in your own materializations.

If so, I don't believe it's accurate to call it the `runtime_config` as it is not an instance of the [RuntimeConfig class in core](https://github.com/dbt-labs/dbt-core/blob/main/core/dbt/config/runtime.py#L90), which is much broader in its scope and responsibilities. 

Perhaps just `materialization_config`? Is it possible to type this input at all through the python implementation?

Reply:
Recapping from 1:1 for visibility - this is a `RuntimeConfigObject` but could just be the `ModelNode` and associated `NodeConfig` to make this interface more narrow in scope. `config.get` does some nice safety handling to ensure there is a config on `config.model` that would be nice to maintain.

Reply:
This has been updated to `config.model`, which is a `ModelNode` most of the time from what I can tell. We only need a `CompiledNode` for a lot of the modeling, and less than that even (`ParsedNode`) in other spots. `adapter.make_materialization_from_runtime_config` will be updated to `adapter.make_materialization_from_node`. This is purposely ambiguous since `config.model` changes meaning over time. But the expectation in our use cases is that we will at least have a `CompiledNode`. Typing will be updated to reflect that `Relation` models need `CompiledNode` instances and `RelationComponent` models will need `ParsedNode` instances.